core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala

```diff
@@ -150,14 +238,99 @@ class InvokerActor extends FSM[InvokerState, InvokerInfo] {
         case Event(StateTimeout, _)   => goto(Offline)
     }
 
+    /**
+     * Handle the completion of an Activation in every state.
+     */
+    whenUnhandled {
+        case Event(cm: InvocationFinishedMessage, info) => handleCompletionMessage(cm.successful, info.buffer)
+    }
+
+    /** Logging on Transition change */
+    onTransition {
+        case _ -> Offline   => transid.mark(this, LoggingMarkers.LOADBALANCER_INVOKER_OFFLINE, s"$name is offline", akka.event.Logging.WarningLevel)
+        case _ -> UnHealthy => transid.mark(this, LoggingMarkers.LOADBALANCER_INVOKER_UNHEALTHY, s"$name is unhealthy", akka.event.Logging.WarningLevel)
+        case _ -> Healthy   => logging.info(this, s"$name is healthy")
+    }
+
+    /** Scheduler to send testActivations, if the invoker is UnHealthy */
     onTransition {
-        case Healthy -> Offline => logging.warn(this, s"$name is down")
-        case Offline -> Healthy => logging.info(this, s"$name is online")
+        case _ -> UnHealthy => {
+            invokeTestAction()
+            setTimer(InvokerActor.timerName, Tick, 1.minute, true)
+        }
+        case UnHealthy -> _ => cancelTimer(InvokerActor.timerName)
     }
 
     initialize()
+
+    /**
+     * Handling for active acks. This method saves the result (successful or unsuccessful)
+     * into an RingBuffer and checks, if the InvokerActor has to be changed to UnHealthy.
+     *
+     * @param wasActivationSuccessful: result of Activation
+     * @param buffer to be used
+     */
+    private def handleCompletionMessage(wasActivationSuccessful: Boolean, buffer: RingBuffer[Boolean]) = {
+        buffer.add(wasActivationSuccessful)
```

Comment:
seems to me you can avoid the ring buffer and just use a saturating counter (reason: you add and count) - i won't hold up the pr for this, but would like to understand if the ring buffer is really necessary.